grudge/discretization.py

```diff
@@ -636,6 +626,133 @@ def normal(self, dd):
     # }}}
 
 
+def make_discretization_collection(
+        array_context: ArrayContext, mesh: Mesh,
```

Comment:
In the long run, that second argument should permit:
* A mesh
* A discretization
* A dictionary from identifiers (like wall/fluid) to either of the above. cc @majosm 

Maybe make document it as positional-only so that we're not tied to its name.

Reply:
@inducer Could you provide a sketch of what you mean?

Reply:
There are two points I'm trying to make here:
* The first one is identical to the comment below. Multi-physics means there's not *the* volume discretization. Instead, there's a map from identifiers to discretizations, with some way to also describe connectivity.
* IMO there's a legitimate reason to want to create a discretization collection if you already *have* a `Discretization`. I.e. it seems like overreach to force the user to provide a mesh and then converting it to a discretization for them.

Reply:
> * IMO there's a legitimate reason to want to create a discretization collection if you already _have_ a `Discretization`. I.e. it seems like overreach to force the user to provide a mesh and then converting it to a discretization for them.

@inducer I cannot think of a current example (using grudge) where any example/experiment explicitly constructs a `Discretization`; it's always exclusively through `DiscretizationCollection`. I agree we should permit flexible arguments to the function. But the heart of my question is more or less what do you think is a good placeholder for the moment? I don't plan to solve this question in this PR.

Reply:
Sticking with a `Mesh` is alright, we can easily disambiguate by type. Just be forward-looking with naming.

Reply:
@inducer Would appreciate your thoughts on https://github.com/inducer/grudge/pull/122/commits/74eb9bb9472ef67e279d7fc02bed71d29711cb8b when you have time. I have to admit, I wasn't entirely sure how to handle this since we have no concrete use cases/examples. So if you had something else in mind, just let me know and I'll update.